crates/loro/tests/loro_rust_test.rs

```diff
@@ -1196,3 +1209,175 @@ fn test_map_checkout_on_trimmed_doc() {
+                    "parent": null,
+                    "meta":{},
+                    "id": "0@0",
+                    "index": 0,
+                    "fractional_index": "80",
+                },
+                {
+                    "parent": "0@0",
+                    "meta":{},
+                    "id": "1@0",
+                    "index": 0,
+                    "fractional_index": "80",
+                },
+                {
+                    "parent": "0@0",
+                    "meta":{},
+                    "id": "3@0",
+                    "index": 1,
+                    "fractional_index": "8180",
+                },
+            ]
+        })
+    );
+    new_doc.checkout(&ID::new(doc.peer_id(), 2).into()).unwrap();
+    assert_eq!(
+        new_doc.get_deep_value(),
+        loro_value!({
+            "tree": [
+                {
+                    "parent": null,
+                    "meta":{},
+                    "id": "0@0",
+                    "index": 0,
+                    "fractional_index": "80",
+                },
+                {
+                    "parent": "0@0",
+                    "meta":{},
+                    "id": "1@0",
+                    "index": 0,
+                    "fractional_index": "80",
+                },
+            ]
+        })
+    );
+    new_doc.checkout(&ID::new(doc.peer_id(), 1).into()).unwrap();
+    assert_eq!(
+        new_doc.get_deep_value(),
+        loro_value!({
+            "tree": [
+                {
+                    "parent": null,
+                    "meta":{},
+                    "id": "0@0",
+                    "index": 0,
+                    "fractional_index": "80",
+                },
+                {
+                    "parent": null,
+                    "meta":{},
+                    "id": "1@0",
+                    "index": 1,
+                    "fractional_index": "8180",
+                },
+            ]
+        })
+    );
+    new_doc.checkout_to_latest();
+    assert_eq!(
+        new_doc.get_deep_value(),
+        loro_value!({
+            "tree": [
+                {
+                    "parent": null,
+                    "meta":{},
+                    "id": "0@0",
+                    "index": 0,
+                    "fractional_index": "80",
+                },
+                {
+                    "parent": "0@0",
+                    "meta":{},
+                    "id": "1@0",
+                    "index": 0,
+                    "fractional_index": "80",
+                },
+                {
+                    "parent": "0@0",
+                    "meta":{},
+                    "id": "3@0",
+                    "index": 1,
+                    "fractional_index": "8180",
+                },
+            ]
+        })
+    );
+
+    let err = new_doc
+        .checkout(&ID::new(doc.peer_id(), 0).into())
+        .unwrap_err();
+    assert_eq!(err, LoroError::SwitchToTrimmedVersion);
+    Ok(())
+}
```

Comment:
I need a test case that works like this:

1. A new doc
2. A bunch of tree ops on the doc
3. A series of other ops on the doc
4. Move the tree nodes on the doc
5. Generate a GC version that starts from 3
6. Import the GC snapshot into a new doc, and switch the new doc to the version 3
7. Check the equality of new doc and the doc at version 3

Reply:
done